evadb/catalog/models/configuration_catalog.py

```diff
@@ -0,0 +1,45 @@
+# coding=utf-8
+# Copyright 2018-2023 EvaDB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from sqlalchemy import Column, String
+
+from evadb.catalog.models.base_model import BaseModel
+from evadb.catalog.models.utils import ConfigurationCatalogEntry, TextPickleType
+
+
+class ConfigurationCatalog(BaseModel):
+    """The `ConfigurationCatalog` catalog stores all the configuration params.
+    `_row_id:` an autogenerated unique identifier.
+    `_key:` the key for the config.
+    `_value:` the value for the config
+    """
+
+    __tablename__ = "configuration_catalog"
+
+    _key = Column("name", String(100), unique=True)
+    _value = Column(
+        "engine", TextPickleType()
+    )  # TODO :- Will this work or would we need to pickle ??
```

Comment:
I had left this comment actually to see if a singleton value like `_value` would get pickled after `json.dumps` that we do in the `TextPickleType()` method. Do check that once if that works.

Also DB column names are `name` and `engine` instead of `key` and `value`

Reply:
Why are you calling them `name` and `engine`?

Reply:
No no, I mean I forgot to update them in code. Just pointing it out so that you change that. :sweat_smile: